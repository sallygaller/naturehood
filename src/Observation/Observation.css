.Observation {
    width: 80%;
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    padding: 15px;
    border-top: 2px solid;
}

.Observation-title {
    font-weight: bold;
    font-family: 'Homemade Apple', cursive;
    padding: 15px;
    font-size: 18px;
}

.Observation-button {
    background-color: #012611;
    width: 20%;
    color: #f2f2f2;
    margin-top: 10px;
    padding: 5px;
    border-radius: 4px;
    cursor: pointer;
}


