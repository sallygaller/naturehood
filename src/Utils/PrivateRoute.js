import React from "react";
import { Route, Redirect } from "react-router-dom";
import TokenService from "../services/token-service";

export default function PrivateRoute({ component, ...props }) {
  const Component = component;
  return (
    <Route
      {...props}
      render={(componentProps) =>
        TokenService.hasAuthToken() ? (
          (console.log({ componentProps }, { props }),
          (<Component {...componentProps} {...props} />))
        ) : (
          <Redirect
            to={{
              pathname: "/login",
              state: { from: componentProps.location },
            }}
          />
        )
      }
    />
  );
}
