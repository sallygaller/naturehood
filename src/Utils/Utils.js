import moment from "moment";

export function dateFormat(observation) {
  const dateFormat = moment(observation.date).format("L");
  return dateFormat;
}

export function timeFormat(observation) {
  const timeFormat = moment(observation.time, "hh:mm:ss").format("hh:mm a");
  return timeFormat;
}

export const speciesTypes = [
  { id: 1, type: "Mammal" },
  { id: 2, type: "Fish" },
  { id: 3, type: "Bird" },
  { id: 4, type: "Arthropod" },
  { id: 5, type: "Amphibian" },
  { id: 6, type: "Reptile" },
];

export const testObservations = [
  {
    id: 1,
    species: "Robin",
    type: "Bird",
    date: "2020-12-25T08:00:00.000Z",
    time: "08:30:00",
    description: "I saw a robin at my feeder",
    lat: 45.593,
    lng: -122.755,
    date_added: "2020-12-25T09:28:32.615Z",
    neighbor: 1,
  },
  {
    id: 2,
    species: "Fox",
    type: "Mammal",
    date: "2020-12-15T08:00:00.000Z",
    time: "04:30:00",
    description: "I saw a mother fox and her cubs. The mother hissed at me.",
    lat: 47.593,
    lng: -122.755,
    date_added: "2020-12-15T16:28:32.615Z",
    neighbor: 1,
  },
  {
    id: 3,
    species: "Raccoon",
    type: "Mammal",
    date: "2020-12-05T08:00:00.000Z",
    time: "23:30:00",
    description: "A group of raccoons chased me down the street.",
    lat: 49.593,
    lng: -122.755,
    date_added: "2020-12-06T09:28:32.615Z",
    neighbor: 1,
  },
  {
    id: 4,
    species: "Skunk",
    type: "Mammal",
    date: "2020-12-08T08:00:00.000Z",
    time: "07:30:00",
    description: "I saw a skunk under my deck.",
    lat: 51.593,
    lng: -123.755,
    date_added: "2020-12-08T10:15:32.615Z",
    neighbor: 1,
  },
];
