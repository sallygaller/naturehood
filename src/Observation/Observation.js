import React from "react";
import { Link } from "react-router-dom";
import { API_ENDPOINT } from "../config";
import moment from "moment";
import "./Observation.css";

export default function Observation(props) {
  const { id, species, date, time, description } = props.observation;
  const dateFormat = moment(date).format("L");
  const timeFormat = moment(time, "hh:mm:ss").format("hh:mm a");
  const timeFormatSplit = timeFormat.split(" ");
  const timeCorrect = timeFormatSplit[0];
  const ampm = timeFormatSplit[1];

  return (
    <div className="Observation">
      <div className="Observation-item Observation-title">{species}</div>
      <div className="Observation-item">
        Date: {dateFormat}
        <br></br>
        Time: {timeCorrect}
        {ampm}
      </div>
      <div className="Observation-item">{description}</div>
      <div className="Observation-item">
        <Link className="Observation-item" to={`/observations/edit/${id}`}>
          <button type="button">Edit</button>
        </Link>
      </div>
      <div className="Observation-item">
        <button onClick={() => props.handleDeleteRequest(id)}>Delete</button>
      </div>
    </div>
  );
}
