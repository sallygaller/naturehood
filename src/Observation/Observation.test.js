import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Observation from "./Observation";
import { observation } from "../Utils/TestFiles";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <Observation observation={observation} />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Observation name="Observation" observation={observation} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders the UI as expected with no observation", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Observation name="Observation" observation={{}} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
