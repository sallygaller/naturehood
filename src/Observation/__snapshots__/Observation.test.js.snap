// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Observation"
>
  <div
    className="Observation-time-date"
  >
    08:30 am
    , 
    12/25/2020
  </div>
  <div
    className="Observation-title"
  >
    Robin
  </div>
  <div>
    I saw a robin at my feeder
  </div>
  <div
    className="Observation-buttons"
  >
    <a
      href="/observations/edit/1"
      onClick={[Function]}
    >
      <button
        className="Observation-button"
        type="button"
      >
        Edit
      </button>
    </a>
     
    <button
      className="Observation-button"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`renders the UI as expected with no observation 1`] = `
<div
  className="Observation"
>
  <div
    className="Observation-time-date"
  >
    Invalid date
    , 
    01/31/2021
  </div>
  <div
    className="Observation-title"
  />
  <div />
  <div
    className="Observation-buttons"
  >
    <a
      href="/observations/edit/undefined"
      onClick={[Function]}
    >
      <button
        className="Observation-button"
        type="button"
      >
        Edit
      </button>
    </a>
     
    <button
      className="Observation-button"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
